data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportScheduler.java

```diff
@@ -64,14 +71,25 @@ public void run() {
                     final ExportPartition exportPartition = (ExportPartition) sourcePartition.get();
                     LOG.info("Acquired an export partition: {}", exportPartition.getPartitionKey());
 
-                    final List<PartitionIdentifier> partitionIdentifiers = mongoDBExportPartitionSupplier.apply(exportPartition);
+                    final String exportPartitionKey = EXPORT_PREFIX + exportPartition.getCollection();
+                    Optional<EnhancedSourcePartition> globalPartition = enhancedSourceCoordinator
+                            .getPartition(exportPartitionKey);
+                    while (globalPartition.isEmpty()) {
+                        LOG.warn("Wait for global partition to be created.");
+                        Thread.sleep(DEFAULT_GET_PARTITION_BACKOFF_MILLIS);
+                        globalPartition = enhancedSourceCoordinator.getPartition(exportPartitionKey);
+                    }
+
+                    final PartitionIdentifierBatch partitionIdentifierBatch = mongoDBExportPartitionSupplier.apply(exportPartition);
 
-                    final boolean createStatus = createDataQueryPartitions(exportPartition.getCollection(), Instant.now(), partitionIdentifiers);
+                    createDataQueryPartitions(
+                            exportPartition.getCollection(), Instant.now(), partitionIdentifierBatch.getPartitionIdentifiers(),
+                            (GlobalState) globalPartition.get());
+                    updateExportPartition(exportPartition, partitionIdentifierBatch);
 
-                    if (createStatus) {
+                    if (partitionIdentifierBatch.isLastBatch()) {
```

Comment:
I assume we still want to check the boolean return value from `createDataQueryPartitions` and call `closeExportPartitionWithError` when false?

Reply:
Is this in the case of a crash and new node coming up? If you want to skip any way you can give up this partition, but maybe I'm misunderstanding.